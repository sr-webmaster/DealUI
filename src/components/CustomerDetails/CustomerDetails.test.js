import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import Component from './CustomerDetails.vue'

describe('Customer Card View', () => {
  let localVue
  describe('Visuals', () => {
    test('defaults', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, { localVue })
      expect(wrapper.element).toMatchSnapshot()
    })
    test('customer property is null', () => {
      localVue = createLocalVue()
      let wrapper = {}
      expect(() => {
        wrapper = mount(Component, {
          localVue,
          propsData: {
            customer: null
          }
        })
      }).not.toThrow()
      expect(wrapper.element).toMatchSnapshot()
    })
    test('customer populated', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue,
        propsData: {
          customer: {
            id: 1,
            first_name: 'First',
            last_name: 'Last',
            email: 'example@email.com',
            phone: '123456789',
            title: 'Sir',
            company: 'Place',
            text_enabled: 'Yes',
            customer_deals: 3
          }
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
    test('rows set', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue,
        propsData: {
          customer: {
            id: 1,
            first_name: 'First',
            last_name: 'Last',
            email: 'example@email.com',
            phone: '123456789',
            title: 'Sir',
            company: 'Place',
            text_enabled: 'Yes',
            customer_deals: 3
          },
          rows: [
            { text: 'Email', key: 'email' },
            { text: 'Phone', key: 'phone' },
            { text: 'Can be texted?', key: 'text_enabled' }
          ]
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe('Computed', () => {
    describe('item', () => {
      test('returns empty object when customer property isNil', () => {
        const wrapper = shallowMount(Component, {
          propsData: {
            customer: null
          }
        })
        expect(wrapper.vm.item).toBeTruthy()
        expect(wrapper.vm.item).toEqual({})
      })
    })
    describe('companyName', () => {
      test('returns string of company name when customer.company is a string', () => {
        const wrapper = shallowMount(Component, {
          propsData: {
            customer: {
              id: 1,
              first_name: 'First',
              last_name: 'Last',
              email: 'example@email.com',
              phone: '123456789',
              title: 'Sir',
              company: 'FreshinUp',
              text_enabled: 'Yes',
              customer_deals: 3
            }
          }
        })
        expect(wrapper.vm.companyName).toEqual('FreshinUp')
      })
      test('returns string of company name when customer.company is an object', () => {
        const wrapper = shallowMount(Component, {
          propsData: {
            customer: {
              id: 1,
              first_name: 'First',
              last_name: 'Last',
              email: 'example@email.com',
              phone: '123456789',
              title: 'Sir',
              company: {
                id: 22,
                uuid: 'c67d7e83-02ee-4c2a-975e-61567c1f4c0a',
                created_at: '2020-04-03 07:19:32',
                status: 1,
                name: 'FreshinUp',
                address: '2266 Test Ave',
                address2: 'unit 5',
                city: 'Madison',
                state: 'WI',
                zip: '53558',
                country: 'US',
                website: null,
                notes: null,
                members_count: null,
                teams_count: null,
                tags: [],
                logo: 'https://via.placeholder.com/800x600.png',
                company_types: []
              },
              text_enabled: 'Yes',
              customer_deals: 3
            }
          }
        })
        expect(wrapper.vm.companyName).toEqual('FreshinUp')
      })
    })
  })
})
