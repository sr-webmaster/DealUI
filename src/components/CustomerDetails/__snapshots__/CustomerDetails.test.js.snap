// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Card View Visuals customer populated 1`] = `
<div
  class="container grid-list-md"
>
  <div
    class="layout row justify-start"
  >
    <div
      class="flex shrink"
    >
      <div
        class="v-avatar secondary"
        style="height: 48px; width: 48px;"
      >
        <span
          class="secondary-text--text"
          style="font-size: 24px;"
        >
          
    FL
  
        </span>
      </div>
    </div>
     
    <div
      class="flex pl-4"
    >
      <h4
        class="title font-weight-bold secondary--text break-word"
      >
        
        First Last
      
      </h4>
       
      <span
        class="v-chip break-word ml-0 v-chip--label theme--light info info-text--text"
        tabindex="0"
      >
        <span
          class="v-chip__content"
        >
          
        ID#: 1
      
        </span>
      </span>
    </div>
  </div>
   
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Email
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          example@email.com
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Phone
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          123456789
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Can be texted?
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          Yes
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Associated Deals
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          3
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Company
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          Place
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Title
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          Sir
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
</div>
`;

exports[`Customer Card View Visuals customer property is null 1`] = `
<div
  class="container grid-list-md"
>
  <div
    class="layout row justify-start"
  >
    <div
      class="flex shrink"
    >
      <div
        class="v-avatar secondary"
        style="height: 48px; width: 48px;"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--light secondary-text--text"
          style="font-size: 48px;"
        >
          account_circle
        </i>
      </div>
    </div>
     
    <div
      class="flex pl-4"
    >
      <h4
        class="title font-weight-bold secondary--text break-word"
      >
        
         
      
      </h4>
       
      <span
        class="v-chip break-word ml-0 v-chip--label theme--light info info-text--text"
        tabindex="0"
      >
        <span
          class="v-chip__content"
        >
          
        ID#: 
      
        </span>
      </span>
    </div>
  </div>
   
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Email
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Phone
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Can be texted?
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Associated Deals
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Company
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Title
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
</div>
`;

exports[`Customer Card View Visuals defaults 1`] = `
<div
  class="container grid-list-md"
>
  <div
    class="layout row justify-start"
  >
    <div
      class="flex shrink"
    >
      <div
        class="v-avatar secondary"
        style="height: 48px; width: 48px;"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--light secondary-text--text"
          style="font-size: 48px;"
        >
          account_circle
        </i>
      </div>
    </div>
     
    <div
      class="flex pl-4"
    >
      <h4
        class="title font-weight-bold secondary--text break-word"
      >
        
         
      
      </h4>
       
      <span
        class="v-chip break-word ml-0 v-chip--label theme--light info info-text--text"
        tabindex="0"
      >
        <span
          class="v-chip__content"
        >
          
        ID#: 
      
        </span>
      </span>
    </div>
  </div>
   
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Email
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Phone
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Can be texted?
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Associated Deals
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Company
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Title
      
    </div>
     
    <div
      class="flex md6 text-xs-right grey--text break-word"
    >
      
          
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
</div>
`;

exports[`Customer Card View Visuals rows set 1`] = `
<div
  class="container grid-list-md"
>
  <div
    class="layout row justify-start"
  >
    <div
      class="flex shrink"
    >
      <div
        class="v-avatar secondary"
        style="height: 48px; width: 48px;"
      >
        <span
          class="secondary-text--text"
          style="font-size: 24px;"
        >
          
    FL
  
        </span>
      </div>
    </div>
     
    <div
      class="flex pl-4"
    >
      <h4
        class="title font-weight-bold secondary--text break-word"
      >
        
        First Last
      
      </h4>
       
      <span
        class="v-chip break-word ml-0 v-chip--label theme--light info info-text--text"
        tabindex="0"
      >
        <span
          class="v-chip__content"
        >
          
        ID#: 1
      
        </span>
      </span>
    </div>
  </div>
   
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Email
      
    </div>
     
    <div
      class="flex md6 text-xs-right undefined break-word"
    >
      
          example@email.com
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Phone
      
    </div>
     
    <div
      class="flex md6 text-xs-right undefined break-word"
    >
      
          123456789
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
  <div
    class="layout row wrap"
  >
    <div
      class="flex font-weight-bold text-uppercase caption md6"
    >
      
        Can be texted?
      
    </div>
     
    <div
      class="flex md6 text-xs-right undefined break-word"
    >
      
          Yes
        
    </div>
  </div>
   
  <hr
    class="my-1 v-divider theme--light"
  />
</div>
`;
